src/Kentico.Xperience.Lucene.Core/Indexing/DefaultLuceneConfigurationStorageService.cs

```diff
@@ -263,6 +287,50 @@ public bool TryEditIndex(LuceneIndexModel configuration)
                     }
                 }
             }
+
+            var removeContentTypesQuery = contentTypeProvider
+                .Get()
+                .WhereIn(nameof(LuceneContentTypeItemInfo.LuceneContentTypeItemIncludedPathItemId), removedPaths.Select(x => x.LuceneIncludedPathItemId).ToArray());
+
+            contentTypeProvider.BulkDelete(removeContentTypesQuery);
+
+            var allExistingContentTypes = await contentTypeProvider
+                .Get()
+                .WhereIn(nameof(LuceneContentTypeItemInfo.LuceneContentTypeItemIncludedPathItemId), configuration.Paths.Select(x => int.Parse(x.Identifier ?? "0")).ToArray())
+                .GetEnumerableTypedResultAsync();
+
+            int[] removedContentTypeIdsFromEditedPaths = allExistingContentTypes
+                .Where(x => !configuration.Paths
+                    .Any(y => y.ContentTypes
+                        .Exists(z => x.LuceneContentTypeItemIncludedPathItemId.ToString() == y.Identifier && x.LuceneContentTypeItemContentTypeName == z.ContentTypeName))
+                )
+                .Select(x => x.LuceneContentTypeItemId)
+                .ToArray();
+
+            contentTypeProvider.BulkDelete(contentTypeProvider.Get().WhereIn(nameof(LuceneContentTypeItemInfo.LuceneContentTypeItemId), removedContentTypeIdsFromEditedPaths));
+
+            foreach (var path in existingPaths)
+            {
+                path.LuceneIncludedPathItemAliasPath = configuration.Paths.Single(x => x.Identifier == path.LuceneIncludedPathItemId.ToString()).AliasPath;
```

Comment:
Do we need `LuceneIndexIncludedPath.Identifier` to be a string? According to usages, we seem to expect this identifier to be an integer.

Reply:
we do not. I have changed it